### PR TITLE
fix(argo-cd): Adding checksum for argocd-cm to dex pod

### PR DESCRIPTION
**Pull request by etiennnr**
Minor fix to restart the dex pod if there is a change to the `argocd-cm."dex.config"` config since [`dex.config`](https://github.com/argoproj/argo-helm/blob/39e1322df2b15446796356d9bf027250c5d75c35/charts/argo-cd/values.yaml#L191C7-L191C17) is stored there.

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

**Review by mbevc1 — APPROVED** [review]

File: charts/argo-cd/Chart.yaml [hunk 1]
```diff
@@ -28,3 +28,5 @@ annotations:
   artifacthub.io/changes: |
     - kind: added
       description: Support ability to set .Values.namespaceOverride
```
Comment (mkilchhofer) [inline charts/argo-cd/Chart.yaml:30]: Can you drop the existing two lines? We only need your change here

**Review by mkilchhofer — CHANGES_REQUESTED** [review]
Almost LGTM

File: charts/argo-cd/Chart.yaml (hunk 1 above)
Comment (mbevc1) [reply charts/argo-cd/Chart.yaml:30]: ah, well spotted! :)

Comment (etiennnr) [reply charts/argo-cd/Chart.yaml]: done :)

**Review by mbevc1 — APPROVED** [review]

**Review by yu-croco — APPROVED** [review]

**Review by mkilchhofer — APPROVED** [review]
Thank you for the contribution 🙏

**Pull request merged**